src/main/java/com/example/librarymanagementsystem/model/Transaction.java

```diff
@@ -0,0 +1,96 @@
+package com.example.librarymanagementsystem.model;
+
+import java.time.LocalDate;
+
+public class Transaction {
+
+    private int transactionId;
+    private int resourceId;
+    private int userId;
+    private LocalDate transactionDate;
+    private LocalDate dueDate;
+    private Status transactionStatus;
+
+
+    /**
+     * Instantiates a new Transaction.
+     *
+     * @param transactionId   the transaction id
+     * @param resourceId      the resource id
+     * @param userId          the user id
+     * @param transactionDate the transaction date
+     * @param dueDate         the due date
+     */
+    public Transaction(int transactionId, int resourceId, int userId, LocalDate transactionDate, LocalDate dueDate) {
+        this.transactionId = transactionId;
+        this.resourceId = resourceId;
+        this.userId = userId;
+        this.transactionDate = transactionDate;
+        this.dueDate = dueDate;
+    }
+
+    /**
+     * Instantiates a new Transaction.
+     *
+     * @param resourceId        the resource id
+     * @param userId            the user id
+     * @param transactionDate   the transaction date
+     * @param dueDate           the due date
+     * @param transactionStatus the transaction status
+     */
+    public Transaction(int resourceId, int userId, LocalDate transactionDate, LocalDate dueDate, Status transactionStatus) {
+        this.resourceId = resourceId;
+        this.userId = userId;
+        this.transactionDate = transactionDate;
+        this.dueDate = dueDate;
+        this.transactionStatus = transactionStatus;
+    }
```

Comment:
Instead of defining strict constructors for other parameters, use builders (`Builder Design Pattern`) to make it more flexible, readable and reducing error encounters. Look into that and see how it works.